src/main/java/io/airlift/slice/SliceUtf8.java

```diff
@@ -428,7 +428,7 @@ public static Slice fixInvalidUtf8(Slice slice, OptionalInt replacementCodePoint
         return utf8.slice(0, utf8Position);
     }
 
-    private static int getCodePointAtSafe(Slice utf8, int position)
+    public static int getCodePointAtSafe(Slice utf8, int position)
```

Comment:
Lets rename this to `tryGetCodePointAt`.  Also, this method will need java docs explaining the return values (the negative number and what they mean).